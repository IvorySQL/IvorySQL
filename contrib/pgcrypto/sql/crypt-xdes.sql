--
-- crypt() and gen_salt(): extended des
--

SET ivorysql.enable_emptystring_to_null to off;
SELECT crypt('', '_J9..j2zz');

SELECT crypt('foox', '_J9..j2zz');

-- check XDES handling of keys longer than 8 chars
SELECT crypt('longlongpassword', '_J9..j2zz');

-- error, salt too short
SELECT crypt('foox', '_J9..BWH');

-- error, count specified in the second argument is 0
SELECT crypt('password', '_........');

-- error, count will wind up still being 0 due to invalid encoding
-- of the count: only chars ``./0-9A-Za-z' are valid
SELECT crypt('password', '_..!!!!!!');

-- count should be non-zero here, will work
SELECT crypt('password', '_/!!!!!!!');

CREATE TABLE ctest (data text, res text, salt text);
INSERT INTO ctest VALUES ('password', '', '');

UPDATE ctest SET salt = gen_salt('xdes', 1001);
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;
RESET ivorysql.enable_emptystring_to_null;

DROP TABLE ctest;
