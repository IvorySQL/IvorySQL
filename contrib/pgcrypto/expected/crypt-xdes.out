--
-- crypt() and gen_salt(): extended des
--
SET ivorysql.enable_emptystring_to_null to off;
SELECT crypt('', '_J9..j2zz');
        crypt         
----------------------
 _J9..j2zzR/nIRDK3pPc
(1 row)

SELECT crypt('foox', '_J9..j2zz');
        crypt         
----------------------
 _J9..j2zzAYKMvO2BYRY
(1 row)

-- check XDES handling of keys longer than 8 chars
SELECT crypt('longlongpassword', '_J9..j2zz');
        crypt         
----------------------
 _J9..j2zz4BeseiQNwUg
(1 row)

-- error, salt too short
SELECT crypt('foox', '_J9..BWH');
ERROR:  invalid salt
-- error, count specified in the second argument is 0
SELECT crypt('password', '_........');
ERROR:  crypt(3) returned NULL
-- error, count will wind up still being 0 due to invalid encoding
-- of the count: only chars ``./0-9A-Za-z' are valid
SELECT crypt('password', '_..!!!!!!');
ERROR:  crypt(3) returned NULL
-- count should be non-zero here, will work
SELECT crypt('password', '_/!!!!!!!');
        crypt         
----------------------
 _/!!!!!!!zqM49hRzxko
(1 row)

CREATE TABLE ctest (data text, res text, salt text);
INSERT INTO ctest VALUES ('password', '', '');
UPDATE ctest SET salt = gen_salt('xdes', 1001);
UPDATE ctest SET res = crypt(data, salt);
SELECT res = crypt(data, res) AS "worked"
FROM ctest;
 worked 
--------
 t
(1 row)

DROP TABLE ctest;
